### PR TITLE
[OPERATOR-511] Migrate portworx pod custom annotations

### PR DESCRIPTION
**Pull request by pure-jliao**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/migration/generate.go [hunk 1]
```diff
@@ -830,6 +835,45 @@ func (h *Handler) addMonitoringSpec(cluster *corev1.StorageCluster, ds *appsv1.D
...
+	)
+	if err != nil {
+		return err
+	}
+
+	customAnnotations := make(map[string]string)
+	for _, pod := range podList.Items {
+		owner := metav1.GetControllerOf(&pod)
+		if owner != nil && owner.UID == ds.UID && pod.Annotations != nil {
+			for k, v := range pod.Annotations {
```
Comment (haibinxie) [inline pkg/migration/generate.go:856]: there is no need to get pod, annotation is in daemonset.

Comment (pure-jliao) [reply pkg/migration/generate.go:856]: I thought user could just annotate the pod manually. so do we look at the daemonset only? or look at both the pods and daemonset? @haibinxie

Comment (haibinxie) [reply pkg/migration/generate.go:856]: the user wont have access to pod when first deploying it. if users change the pod later directly, then when they upgrade portworx (px version or other stuff), the annotation will be changed back during daemonset rolling upgrade. In any case, I think it's correct to refer to pod as source of truth. so let's keep it unchanged.

**Review by haibinxie — APPROVED** [review]

File: pkg/migration/generate.go (hunk 1 above)
Comment (piyush-nimbalkar) [reply pkg/migration/generate.go:856]: I agree with Michael about just checking daemonset. If customers are adding annotation directly to pods, then it is not our responsibility to copy them as they are anyway ephemeral. When the DS is updated, the annotations are going to go away. We should only pick up the persisted annotations from the DS.

File: pkg/migration/generate.go [hunk 2]
```diff
@@ -830,6 +835,45 @@ func (h *Handler) addMonitoringSpec(cluster *corev1.StorageCluster, ds *appsv1.D
...
+				if k != constants.AnnotationPodSafeToEvict {
```
Comment (piyush-nimbalkar) [inline pkg/migration/generate.go:857]: Almost all spec generator daemonsets have this annotation - `portworx.com/install-source`. Let's skip that too.

**Pull request merged**